### ISSUE TITLE
torch.nn.functional import grid_sample

### ISSUE THREAD

**Issue by IlichevSergey** [pytorch/pytorch#33047]
If you have a question or would like help and support, please ask at our
[forums](https://discuss.pytorch.org/).

If you are submitting a feature request, please preface the title with [feature request].
If you are submitting a bug report, please fill in the following details.

## Issue description

torch.nn.functional.grid_sample() behavior is differ from pytorch 1.2.0

## Code example

pytorch 1.2.0:
grid_sample(input=inp, grid=grid, mode='bilinear', padding_mode='border')

pytorch 1.4.0:
the correct way to reproduce behavior of 1.2.0 is:
grid_sample(input=inp, grid=grid, mode='bilinear', padding_mode='border', align_corners=True)
but the default value of align_corners is None, so that produces some errors when update to pytorch 1.4 from 1.2

## System Info

PyTorch 1.2.0:

Collecting environment information...
PyTorch version: 1.2.0
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: Microsoft Windows 10 Enterprise
GCC version: (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0
CMake version: version 3.15.5

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.1.105
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 418.96
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] pytorch-memlab==0.0.2
[pip] torch==1.2.0
[pip] torchvision==0.4.0
[conda] blas                      1.0                         mkl
[conda] libmklml                  2019.0.3                      0
[conda] mkl                       2019.5                      281    anaconda
[conda] mkl-service               2.3.0            py36hb782905_0    anaconda
[conda] mkl_fft                   1.0.12           py36h14836fe_0
[conda] mkl_random                1.0.2            py36h343c172_0
[conda] pytorch                   1.2.0           py3.6_cuda100_cudnn7_1    pytorch
[conda] pytorch-memlab            0.0.2                    pypi_0    pypi
[conda] torchvision               0.4.0                py36_cu100    pytorch

PyTorch 1.4.0:
Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 10 Enterprise
GCC version: (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0
CMake version: version 3.15.5

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.1.105
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 418.96
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] pytorch-memlab==0.0.2
[pip] torch==1.4.0
[pip] torchvision==0.5.0
[conda] blas                      1.0                         mkl
[conda] libmklml                  2019.0.3                      0
[conda] mkl                       2019.5                      281    anaconda
[conda] mkl-service               2.3.0            py36hb782905_0    anaconda
[conda] mkl_fft                   1.0.12           py36h14836fe_0
[conda] mkl_random                1.0.2            py36h343c172_0
[conda] pytorch                   1.4.0           py3.6_cuda101_cudnn7_0    pytorch
[conda] pytorch-memlab            0.0.2                    pypi_0    pypi
[conda] torchvision               0.5.0                py36_cu101    pytorch

**Comment by fmassa** [comment]
Can you please post a snippet reproducing the issue?

**Comment by IlichevSergey** [comment]
This code produces different results in 1.2.0 and 1.4.0 versions of PyToch

import torch
import numpy as np
from torch.nn.functional import grid_sample

torch.manual_seed(0)
np.random.seed(1)
inp = torch.tensor(np.float32(np.random.random_sample((1, 1, 720, 1280))), device='cuda' )
grid = torch.tensor(np.float32(np.random.random_sample((1, 720, 1280, 2))), device='cuda' )
result = grid_sample(input=inp, grid=grid, mode='bilinear', padding_mode='border')

Well, I see that there is a warning about this

**Comment by jasonkena** [comment]
Are you sure you used the same version of Numpy? Numpy's random number generator might have changed.

**Comment by taras-janea** [comment]
@IlichevSergey that's an expected behavior: default `grid_sample` and `affine_grid` behavior has changed to `align_corners=False` since 1.3.0. 
If the old behavior is desired, then `align_corners=True` must be specified. For more information, see [grid_sample](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html) documentation.
